sys/include/net/tcp.h

```diff
@@ -72,9 +72,19 @@ typedef struct __attribute__((packed)) {
 typedef struct __attribute__((packed)) {
     uint8_t kind;     /**< TCP options "Kind" field */
     uint8_t length;   /**< TCP options "Length" field */
-    uint8_t value[];  /**< TCP options "Value" field */
 } tcp_hdr_opt_t;
 
+/**
+ * @brief   Get the start of the TCP option value
+ *
+ * @param[in]   opt   TCP option field helper structure contiguous memory
+ *
+ * @returns     pointer to first byte of the value
+ */
+inline static uint8_t *tcp_hdr_opt_value_ptr(tcp_hdr_opt_t *opt){
```

Comment:
Why is this static? I don't see any point in having a static function outside of a compilation unit.

Reply:
Making inline functions non-static will cause the linker to fail once you include `net/tcp.h` in multiple c-files ;-).

Reply:
Okay. I just don't understand why. Inlining should be enough to deal with linkage issues.